File: java/socketd/src/main/java/org/noear/socketd/core/SimpleListener.java
package org.noear.socketd.core;

import java.io.IOException;

/**
 * 简单的监听器（一般用于占位）
 *
 * @author noear
 * @since 2.0
 */
public class SimpleListener implements Listener {
    /**
     * 打开时
     *
     * @param session 会话
     */
    @Override
    public void onOpen(Session session) throws IOException{

    }

    /**
     * 收到消息时
     *
     * @param session 会话
     * @param message 消息
     */
    @Override
    public void onMessage(Session session, Message message) throws IOException {

    }

    /**
     * 关闭时
     *
     * @param session 会话
     */
    @Override
    public void onClose(Session session) {

    }

    /**
     * 出错时
     *
     * @param session 会话
     * @param error   错误信息
     */
    @Override
    public void onError(Session session, Throwable error) {

    }
}

File: swift/socketd/socketd/src/socketd/transport/core/listener/SimpleListener.swift
//
//  SimpleListener.swift
//  socketd
//
//  Created by noear on 2023/12/28.
//

import Foundation

/**
 * 简单监听器（一般用于占位）
 *
 * @author noear
 * @since 2.0
 */
class SimpleListener : Listener{
    func onOpen(_ session: Session) {
        
    }
    
    func onMessage(_ session: Session, _ message: Message) {
        
    }
    
    func onClose(_ session: Session) {
        
    }
    
    func onError(_ session: Session, _ error: Error) {
        
    }
    
    
}
